### PR TITLE
[3.x] Refactoring MailEclipse Package

### PR DESCRIPTION
**Pull request by ReeceM**
This PR will contain the initial set of refactoring changes that would encourage an easier on-boarding of new contributors. 

It will also hopefully make it easier to get unit tests going on the project so that the state is known when new people add/remove things :)

Based on #129 

Will be a draft for now as the new edits are made so that the PR isn't merged, any new updates to master I will add to my branch, so please feel free to continue accepting PRs if there are any.

### CODE REVIEW

File: src/MailEclipse.php [hunk 1]
```diff
@@ -255,13 +258,13 @@ public static function createTemplate($request): JsonResponse
...
 
-            $dir = resource_path('views/vendor/'.self::$view_namespace.'/templates');
+            $dir = resource_path('views/vendor/'.self::VIEW_NAMESPACE.'/templates');
 
             if (! File::isDirectory($dir)) {
                 File::makeDirectory($dir, 0755, true);
             }
 
-            file_put_contents($dir."/{$templateName}.blade.php", self::templateComponentReplace($request->content));
+            file_put_contents($dir."/{$templatename}.blade.php", Replacer::toBlade($request->content));
```
Comment (ReeceM) [inline src/MailEclipse.php:267]: Darn, Shall fix this... merge issue @ReeceM

**Review by ReeceM — COMMENTED** [review]
Some fixes still need to be changed, one issue found in the way of the merge from master

**Comment by ReeceM** [comment]
Actually, looking at all the changes being made, I could split this into a bunch of PR's

**Comment by ReeceM** [comment]
Just a note so that it isn't assumed it's abandoned, currently just handling a few things and will be able to come back to this update. 

As I do feel it should help add new features as contributors can be given a clear idea as to where to add features

**Review by Qoraiche — APPROVED** [review]
Hello @ReeceM,

it looks good to me, it can be merged

thank you for your contribution,

**Comment by ReeceM** [comment]
Cool \:)

I'll merge merge it to master for testing purposes, I do think I may have broken stuff. So I won't tag anything till it's fixed and running smooth.

**Pull request merged**